src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs

```diff
@@ -189,6 +190,41 @@ public static void DefaultSettings_Echo_RoundTrips_String_Streamed_Async_WithSin
         Assert.True(success, "Test Scenario: DefaultSettings_Echo_RoundTrips_String_Streamed_Async_WithSingleThreadedSyncContext timed-out.");
     }
 
+    [Fact]
+    [OuterLoop]
+    public static void ServiceContract_EchoMessageParameter()
+    {
+        // This test verifies the scenario where MessageParameter attribute is used in the contract
+        StringBuilder errorBuilder = new StringBuilder();
+        try
+        {
+            // *** SETUP *** \\
+            CustomBinding customBinding = new CustomBinding();
+            customBinding.Elements.Add(new TextMessageEncodingBindingElement());
+            customBinding.Elements.Add(new HttpTransportBindingElement());
+
+            // Note the service interface used.  It was manually generated with svcutil.
+            ChannelFactory<IWcfServiceGenerated> factory = new ChannelFactory<IWcfServiceGenerated>(customBinding, new EndpointAddress(Endpoints.DefaultCustomHttp_Address));
+            IWcfServiceGenerated serviceProxy = factory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            string result = serviceProxy.EchoMessageParameter("Hello");
+            if (!string.Equals(result, "Hello Hello"))
+            {
+                errorBuilder.AppendLine(String.Format("Expected response from Service: {0} Actual was: {1}", "Hello", result));
+            }
```

Comment:
Confused here. we expect "Hello" back or "Hello Hello" back? 
*TINY* NIT: Consider putting the echo string into a variable so we don't have too many magic strings

Why can't we just use `Assert.True(string.Equals(result, "Hello", "Expected response from service..."));`